rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/interactor/CovidPatientHistoryActivityInteractorTest.java

```diff
@@ -0,0 +1,140 @@
+    private static final String PATIENT_KEY = "test_patient";
+    private static final String PATIENT_VALUE = "Test Patient";
+
+    private CovidPatientHistoryActivityInteractor interactor;
+
+    private PatientHistoryRepository patientHistoryRepository;
+    private Map<String, String> formWidgetKeyToTextMap;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        interactor = new CovidPatientHistoryActivityInteractor();
+
+        EventClientRepository repository = new EventClientRepository();
+        Event event = repository.convert(EVENT_JSON, Event.class);
+        EventClient eventClient = new EventClient(event);
+
+        patientHistoryRepository = Mockito.mock(PatientHistoryRepository.class);
+        ReflectionHelpers.setField(interactor, "patientHistoryRepository", patientHistoryRepository);
+
+        Mockito.doReturn(eventClient).when(patientHistoryRepository).getEvent(ArgumentMatchers.anyString(), ArgumentMatchers.anyString(), ArgumentMatchers.anyString());
+
+        formWidgetKeyToTextMap = FormKeyTextExtractionUtil.getFormWidgetKeyToTextMap();
+    }
+
+    @Test
+    public void testGetPatientHistoryShouldReturnListOfPatientEntries() throws Exception {
+        List<PatientHistoryEntry> list = interactor.getPatientHistoryEntries("", "", "");
+
+        Assert.assertNotNull(list);
+
+        for (PatientHistoryEntry entry : list) {
+            Assert.assertNotEquals("", entry.getValue());
+            if ("Dimana assessment ini dilakukan?".equals(entry.getKey())) {
```

Comment:
Could we create a key-value map instead and verify the entries match the map?

Reply:
fixed.